Lib/profiling/sampling/gecko_collector.py

```diff
@@ -0,0 +1,451 @@
+import json
+import os
+import platform
+import sys
+import time
+from typing import Dict, List, Optional, Tuple, Any
+
+from .collector import Collector, THREAD_STATE_RUNNING
+
+
+class GeckoCollector(Collector):
+    def __init__(self, *, skip_idle=False):
```

Comment:
ℹ️ **Style**

Consider adding type hints for the `__init__` method parameters to improve code readability and maintainability.

**Suggested fix:**
```suggestion
def __init__(self, *, skip_idle: bool = False):
```